internal/storage/image.go

```diff
@@ -445,6 +463,168 @@ func (svc *imageService) imageStatus(systemContext *types.SystemContext, unstabl
+	}
+	log.Debugf(ctx, "Local manifest digest for image %q is: %s", imageParsed, localManifestDigest)
+
+	canonicalRef, err := reference.WithDigest(reference.TrimNamed(imageParsed), localManifestDigest)
+	if err != nil {
+		return "", fmt.Errorf("create canonical reference: %w", err)
+	}
+
+	return canonicalRef.String(), nil
+}
+
+func (svc *imageService) pullLocalManifestDigest(ctx context.Context, sys *types.SystemContext, policyContext *signature.PolicyContext, imageParsed reference.Named) (digest.Digest, error) {
+	srcRef, err := docker.NewReference(imageParsed)
+	if err != nil {
+		return "", fmt.Errorf("creating docker:// reference for %q: %w", imageParsed.String(), err)
+	}
+
+	destRef, err := istorage.Transport.NewStoreReference(svc.store, imageParsed, "")
+	if err != nil {
+		return "", fmt.Errorf("creating containers-storage: reference for %q: %w", imageParsed.String(), err)
+	}
+
+	manifestBytes, err := copy.Image(ctx, policyContext, destRef, srcRef, &copy.Options{
+		SourceCtx:      sys,
+		DestinationCtx: sys,
+	})
+	if err != nil {
+		var e errcode.Error
+		if as := errors.As(err, &e); as && e.ErrorCode().Descriptor().HTTPStatusCode == http.StatusUnauthorized {
+			// We cannot support private registries right now, but may with future support from the CRI.
+			log.Errorf(ctx, "Verifying policies against private registries it not supported yet: %v", err)
+			return "", nil
+		}
+
+		var policyErr signature.PolicyRequirementError
+		if errors.As(err, &policyErr) {
+			return "", fmt.Errorf("%w: %w", crierrors.ErrSignatureValidationFailed, err)
+		}
+
+		return "", fmt.Errorf("pulling %q to %q: %w", transports.ImageName(srcRef), transports.ImageName(destRef), err)
+	}
+	return digest.FromBytes(manifestBytes), nil
+}
+
+func (svc *imageService) IsRunningImageAllowed(ctx context.Context, systemContext *types.SystemContext, imageName RegistryImageReference, localManifestDigest digest.Digest) error {
+	policy, err := signature.DefaultPolicy(systemContext)
+	if err != nil {
+		return fmt.Errorf("get default policy: %w", err)
+	}
+
+	policyContext, err := signature.NewPolicyContext(policy)
+	if err != nil {
+		return fmt.Errorf("create policy context: %w", err)
+	}
+
+	defer func() {
+		if err := policyContext.Destroy(); err != nil {
+			log.Errorf(ctx, "error destroying policy: %+v", err)
+		}
+	}()
+
+	if err := svc.checkSignature(ctx, systemContext, policyContext, imageName, localManifestDigest); err != nil {
+		return fmt.Errorf("checking signature of %q: %w", imageName, err)
+	}
+
+	log.Debugf(ctx, "Is allowed to run config image %s (policy path: %q)", imageName, systemContext.SignaturePolicyPath)
+
+	return nil
+}
+
+func (svc *imageService) checkSignature(ctx context.Context, sys *types.SystemContext, policyContext *signature.PolicyContext, imageName RegistryImageReference, localManifestDigest digest.Digest) error {
+	imageParsed, err := reference.ParseNormalizedNamed(imageName.Raw().String())
```

Comment:
Why is this round-tripping through a string?!
```suggestion
	imageParsed, err := reference.ParseNormalizedNamed(imageName.Raw().String())
```

Reply:
@mtrmac I changed it from `imageName.Raw().String())`  to `imageName.StringForOutOfProcessConsumptionOnly()`   

Is this a better approach?

Reply:
No, it should not be using a string _at all_. Just use `imageName.Raw()` directly, what is the benefit of the `ParseNormalizedNamed` round-trip?

Reply:
And, uh, I was rather serious about the `ForOutOfProcessConsumptionOnly` name.

Reply:
>No, it should not be using a string at all. Just use imageName.Raw() directly, what is the benefit of the ParseNormalizedNamed round-trip?

Aha, got it! Thanks.

Reply:
> And, uh, I was rather serious about the `ForOutOfProcessConsumptionOnly` name.

:D